culinary-canvas/capstone-project/frontend/my_capstone_project/src/FindRecipes.jsx

```diff
@@ -0,0 +1,139 @@
+
+  const handleInputChange = (index, value) => {
+    const newIngredients = [...ingredients];
+    newIngredients[index] = value;
+    setIngredients(newIngredients);
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    setLoading(true);
+    try {
+      const response = await fetch("http://localhost:3000/recommendations", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({ userId: user.id, ingredients }),
+      });
+      if (response.ok) {
+        setLoading(false);
+        const data = await response.json();
+        setRecipes(data);
+      }
+    } catch (error) {
+      console.error("Error fetching recipes: ", error);
+    }
+  };
+
+  const handleSaveRecipe = async (recipe) => {
+    try {
+      const response = await fetch("http://localhost:3000/recipes/saved", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({
+          userId: user.id,
+          recipeId: recipe.id,
+          recipeName: recipe.label,
+          recipeImage: recipe.image,
+        }),
+      });
+      if (response.ok) {
+        alert("Recipe saved successfully!");
+      }
+    } catch (error) {
+      console.error("Error saving recipes: ", error);
+    }
+  };
+
+  const handleLikeRecipe = async (recipe) => {
+    try {
+      const response = await fetch("http://localhost:3000/recipes/like", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({
+          userId: user.id,
+          recipeId: recipe.id,
+          recipeName: recipe.label,
+          recipeImage: recipe.image,
+        }),
+      });
+      if (response.ok) {
+        alert("Recipe liked successfully!");
+      }
+    } catch (error) {
+      console.error("Error fetching recipe: ", error);
+      console.log(error);
+    }
+  };
+  return (
+    <div>
+      <Link to="/">
+        <button>Back to Home Page</button>
+      </Link>
+      <Link to="/login/:id">
+        <button>Back to user's page</button>
+      </Link>
+      <div className="find-recipes-container">
+        <form onSubmit={handleSubmit}>
+          <h2>Find Recipes</h2>
+          <div>
+            <label>
+              Ingredient 1:
+              <input
+                type="text"
+                name="ingredient1"
+                value={ingredients[0]}
+                onChange={(e) => handleInputChange(0, e.target.value)}
+              />
+            </label>
+          </div>
+          <div>
+            <label>
+              Ingredient 2:
+              <input
+                type="text"
+                name="ingredient2"
+                value={ingredients.ingredient2}
+                onChange={(e) => handleInputChange(1, e.target.value)}
+              />
+            </label>
+          </div>
+          <div>
+            <label>
+              Ingredient 3:
```

Comment:
This is good, but feels like the user is limited to only 3 ingredients? Is it possible to add support for arbitrary `N` ingredients (e.g. `N=4, N=5, ...`)?

Maybe there can be a button (like a plus button or a button that says `Add search ingredient`) and it dynamically adds ingredients that the user can search by? For example see these images:

Default:
![Screenshot 2024-07-08 at 10 40 57 PM](https://github.com/OluwadaraO/Capstone-project-/assets/139017143/6ccdfdea-e871-4e6d-85af-1681b1c6a03f)

Clicking once:
![Screenshot 2024-07-08 at 10 41 08 PM](https://github.com/OluwadaraO/Capstone-project-/assets/139017143/aa109657-adda-4b5f-a89e-0b95c6daf9be)

Clicking again:
![Screenshot 2024-07-08 at 10 41 16 PM](https://github.com/OluwadaraO/Capstone-project-/assets/139017143/dc1a0506-7978-4b50-ab2e-f661cef438eb)

Of course this can be tackled as part of a stretch goal.